x-pack/plugins/uptime/public/components/common/charts/duration_chart.tsx

```diff
@@ -42,20 +41,34 @@ interface DurationChartProps {
   anomalies: AnomalyRecords | null;
 }
 
+export const DurationChart: React.FC<DurationChartProps> = (props) => {
+  const { from, to, updateDateRange } = useAbsoluteDateRange();
+  return (
+    <DurationChartComponent max={to} min={from} {...props} updateDateRange={updateDateRange} />
+  );
+};
+
```

Comment:
i don't understand the need to create this new component?

Reply:
My goal was to avoid relying on references to the react-redux `Provider`, to help keep unit tests cleaner. `useAbsoluteDateRange` relies on `useSelector` and `useDispatch`.

My thinking was that by adding this component, the external API for the component doesn't change (or becomes simpler in some cases) and the complexity required to test the novel functionality doesn't increase.

Reply:
ahh makes sense in that case.